Products/ZenCollector/scheduler.py

```diff
@@ -154,7 +157,14 @@ def late(self):
         """
         Called whenever this task is late and missed its scheduled run time.
         """
-        # TODO: report an event
+        try:
+            # send event only for missed runs on devices.
+            self.task._eventService.sendEvent(
+                STATUS_EVENT, device=self.task._devId,
+                summary="Missed run on {}".format(self.task._manageIp),
+                component="zencommand", severity=Event.Warning)
```

Comment:
The event isn't specific enough because a device can have many tasks, especially in collectors like zenpython.  Something to specifically identify this task needs to be added to the event, probably added to the eventkey. An event class should be specified as well.

A corresponding clear event is not sent when the task is back on time.